addons/binding/org.openhab.binding.bluetooth.ruuvitag/src/main/java/org/openhab/binding/bluetooth/ruuvitag/internal/RuuviTagDiscoveryParticipant.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bluetooth.ruuvitag.internal;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.bluetooth.BluetoothBindingConstants;
+import org.openhab.binding.bluetooth.BluetoothDevice;
+import org.openhab.binding.bluetooth.discovery.BluetoothDiscoveryParticipant;
+import org.openhab.binding.bluetooth.ruuvitag.RuuviTagBindingConstants;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * This discovery participant is able to recognize ruuvitag devices and create discovery results for them.
+ *
+ * @author Sami Salonen- Initial contribution
+ *
+ */
+@Component(immediate = true)
+public class RuuviTagDiscoveryParticipant implements BluetoothDiscoveryParticipant {
```

Comment:
Add `@NonNullByDefault` as annotation to this class and remove the `@NonNull` annotations.

Reply:
Thanks -- should have remembered that...Fixed in 7feb41a